### DISCUSSION TITLE
Visual Studio CodeQL Extension - Downloading database from GitHub

### DISCUSSION THREAD

**Discussion by snc-howell-test** [github/codeql#17067] [Q&A]
Hello, I am running the following:

VS Code Version: 1.91.1 (Universal) on MacOS 
CodeQL VS Code Extension: v1.13.1
CodeQL CLI: 2.17.0

I am part of a GitHub Enterprise Account with a GitHub Advanced Security seats. We have configured CodeQL for our repositories. I see findings populating in the "Security" -> "Code scanning" page of the repos "Security Overview". I would like to download CodeQL databases from our enterprise repo. Downloading a publicly available repo such as apache kafka works as expected: 
<img width="1439" alt="image" src="https://github.com/user-attachments/assets/010c3906-7508-4a7c-8289-c10b74d70268">
<img width="1440" alt="image" src="https://github.com/user-attachments/assets/5d5ce55f-0698-4482-98b4-42dbde5d8ec2">

However, when I attempt to download a database from my enterprise repo I receive an error that it does not exists
<img width="1438" alt="image" src="https://github.com/user-attachments/assets/5ccd56d4-4206-4286-92ce-a53729111964">
<img width="1438" alt="image" src="https://github.com/user-attachments/assets/07900f22-4738-4c31-a276-2e859b4872a7">

This response leads me to believe I'm missing credentials somewhere. The "Account" section in the bottom right empty and does not give me an option to sign in to github when I have the codeQL extension selected. Other extensions such as the GitHub Actions and GitHub Codespaces allow me to sign in. Even after signing into github through the Codespaces extension, the database download for my enterprise repo fails. When I am signed in, I can view trusted extensions. Should I. expect to see the CodeQL extension here as well? 

<img width="1436" alt="image" src="https://github.com/user-attachments/assets/a9764e55-1dcc-4815-ae26-857048499d76">

Thank you all for your help!

**Comment by mbg** [comment]
Hi @snc-howell-test 👋🏻 

Thanks for the questions! 

> This response leads me to believe I'm missing credentials somewhere

In the command palette, there is an option for "CodeQL: Authenticate to GitHub". Have you tried this and logged in with the right account there?

> When I am signed in, I can view trusted extensions. Should I. expect to see the CodeQL extension here as well?

I believe that this isn't about access to credentials, but rather whether extensions are run when you open a workspace and choose the "I do not trust the authors of this project" option.

**Comment by snc-howell-test** [comment]
@mbg Thank you for your response. I do not see an option for "CodeQL: Authenticate to GitHub" in the command palette. 
<img width="645" alt="image" src="https://github.com/user-attachments/assets/3e22a273-5e87-40db-bd41-b4e4e6a35d1f">

Extension trust with VS Code may be managed by our organization. I will follow up with the IT department.

**Reply by mbg** [reply]
Sorry about that. It seems like that option isn't generally available. 

There is a setting where you can specify a custom GitHub Enterprise Server URI:

![image](https://github.com/user-attachments/assets/7e7fa335-f2fc-4bb8-bfa9-19435949b74c)

Or, alternatively, this corresponds to the `"github-enterprise.uri": ""` setting in the JSON configuration.
